taosWS/statement.go

```diff
@@ -0,0 +1,517 @@
+
+	"github.com/taosdata/driver-go/v3/common"
+	"github.com/taosdata/driver-go/v3/common/param"
+	"github.com/taosdata/driver-go/v3/common/serializer"
+	stmtCommon "github.com/taosdata/driver-go/v3/common/stmt"
+	"github.com/taosdata/driver-go/v3/types"
+)
+
+type Stmt struct {
+	stmtID        uint64
+	conn          *taosConn
+	buffer        bytes.Buffer
+	pSql          string
+	isInsert      bool
+	cols          []*stmtCommon.StmtField
+	colTypes      *param.ColumnType
+	queryColTypes []*types.ColumnType
+}
+
+func (stmt *Stmt) Close() error {
+	err := stmt.conn.stmtClose(stmt.stmtID)
+	stmt.buffer.Reset()
+	stmt.conn = nil
+	return err
+}
+
+func (stmt *Stmt) NumInput() int {
+	if stmt.colTypes != nil {
+		return len(stmt.cols)
+	}
+	return -1
+}
+
+func (stmt *Stmt) Exec(args []driver.Value) (driver.Result, error) {
+	if stmt.conn == nil {
+		return nil, driver.ErrBadConn
+	}
+	if len(args) != len(stmt.cols) {
+		return nil, fmt.Errorf("stmt exec error: wrong number of parameters")
+	}
+	block, err := serializer.SerializeRawBlock(param.NewParamsWithRowValue(args), stmt.colTypes)
+	if err != nil {
+		return nil, err
+	}
+	err = stmt.conn.stmtBindParam(stmt.stmtID, block)
+	if err != nil {
+		return nil, err
+	}
+	err = stmt.conn.stmtAddBatch(stmt.stmtID)
+	if err != nil {
+		return nil, err
+	}
+	affected, err := stmt.conn.stmtExec(stmt.stmtID)
+	if err != nil {
+		return nil, err
+	}
+	return driver.RowsAffected(affected), nil
+}
+
+func (stmt *Stmt) Query(args []driver.Value) (driver.Rows, error) {
+	if stmt.conn == nil {
+		return nil, driver.ErrBadConn
+	}
+	block, err := serializer.SerializeRawBlock(param.NewParamsWithRowValue(args), param.NewColumnTypeWithValue(stmt.queryColTypes))
+	if err != nil {
+		return nil, err
+	}
+	err = stmt.conn.stmtBindParam(stmt.stmtID, block)
+	if err != nil {
+		return nil, err
+	}
+	err = stmt.conn.stmtAddBatch(stmt.stmtID)
+	if err != nil {
+		return nil, err
+	}
+	_, err = stmt.conn.stmtExec(stmt.stmtID)
+	if err != nil {
+		return nil, err
+	}
+	return stmt.conn.stmtUseResult(stmt.stmtID)
+}
+
+func (stmt *Stmt) CheckNamedValue(v *driver.NamedValue) error {
+	if stmt.isInsert {
+		if stmt.cols == nil {
+			cols, err := stmt.conn.stmtGetColFields(stmt.stmtID)
+			if err != nil {
+				return err
+			}
+			colTypes := make([]*types.ColumnType, len(cols))
+			for i, col := range cols {
+				t, err := col.GetType()
+				if err != nil {
+					return err
+				}
+				colTypes[i] = t
+			}
+			stmt.cols = cols
+			stmt.colTypes = param.NewColumnTypeWithValue(colTypes)
+		}
+		if v.Ordinal > len(stmt.cols) {
+			return nil
+		}
+		if v.Value == nil {
+			return nil
+		}
+		switch stmt.cols[v.Ordinal-1].FieldType {
+		case common.TSDB_DATA_TYPE_NULL:
+			v.Value = nil
+		case common.TSDB_DATA_TYPE_BOOL:
+			rv := reflect.ValueOf(v.Value)
+			switch rv.Kind() {
+			case reflect.Bool:
+				v.Value = types.TaosBool(rv.Bool())
+			case reflect.Float32, reflect.Float64:
+				v.Value = types.TaosBool(rv.Float() == 1)
```

Comment:
why not to use > 0?

Reply:
"1" represents true in the strconv.ParseBool method. It can also be changed to > 0 for true